### DISCUSSION TITLE
Video within the UIWindow / Scrollable window elements

### DISCUSSION THREAD

**Discussion by fregapple** [MyreMylar/pygame_gui#251] [General]
Hey, 

been playing around to see if it was possible to run video output into a UIWindow.  I can't see any information on it, so not sure where I'd look / if it were to be possible!

**Comment by MyreMylar** [comment]
There isn't default support for it because it would drag in a whole chain
of other video processing libraries and it's probably drifting a bit too
far away from the core purpose of GUI stuff for games - however there is a
library recently making the rounds on Reddit to play videos on a pygame
surface and I expect it would be reasonably easy to make that surface the
background of a UIWindow.

There is a fairly old example script in pygame_gui_examples that plays two
instances of pong in separate UIWindows operating on similar principles.

I expect there would be more work to do if you wanted the windows + video
to be dynamically resizable but it would be a decent starting point.

**Reply by fregapple** [reply]
Nice, I'll search around reddit for the library! I have been looking at the pong example and was attempting to migrate it to my use case, however I wasn't able to get it to work. But it could be I need to tinker with it a bit more.

Speaking of UIWindows / UIPanels, is there a way to make them scrollable? For example an indepth settings list that won't fit on the screen, especially if resized smaller. Having either a fixed pane that can scroll through other UIElements, or a UIWindow that can do the same would be interesting.

**Comment by fregapple** [comment]
Hey!

After souring through the wiki I found this - https://pygame-gui.readthedocs.io/en/latest/theme_reference/theme_scrolling_container.html.

Which is what I am looking for, however I can't workout how I implement it. There is no example of a scrolling container so I can't copy any code. 

I did come across more documentation with:
![image](https://user-images.githubusercontent.com/82219216/152460183-9c97bc4d-c132-448e-8356-d4b166461707.png)

Which I have implemented a window with a scrollbar and a button. But how would I go about moving the contents of the window down with the scroll bar. 

![image](https://user-images.githubusercontent.com/82219216/152460257-77bc2afa-8d56-41a3-8264-c328fa5beca2.png)

**Comment by MyreMylar** [comment]
There is an example of using a scrolling container in my pygame paint demo
here:

https://github.com/MyreMylar/pygame_paint/blob/master/ui/ui_canvas_window.py

**Comment by fregapple** [comment]
That is exactly what I am looking for thankyou!
